package/src/types.ts

```diff
@@ -0,0 +1,103 @@
+			 *
+			 * @param {object} params
+			 * @param {string} params.name
+			 * @param {string} params.content
+			 *
+			 * @see https://astro-integration-kit.netlify.app/utilities/add-virtual-import/
+			 *
+			 * @example
+			 * ```ts
+			 * // my-integration/index.ts
+			 * import { addVirtualImport } from "astro-integration-kit";
+			 *
+			 * addVirtualImport(
+			 *   name: 'virtual:my-integration/config',
+			 *   content: `export default ${ JSON.stringify({foo: "bar"}) }`,
+			 * );
+			 * ```
+			 *
+			 * This is then readable anywhere else in your integration:
+			 *
+			 * ```ts
+			 * // myIntegration/src/component/layout.astro
+			 * import config from "virtual:my-integration/config";
+			 *
+			 * console.log(config.foo) // "bar"
+			 * ```
+			 */
+			addVirtualImport: (params: { name: string; content: string }) => void;
+			/**
+			 * Adds a [vite plugin](https://vitejs.dev/guide/using-plugins) to the
+			 * Astro config.
+			 *
+			 * @param {Params} params
+			 * @param {import("vite").Plugin} params.plugin
+			 * @param {import("astro").HookParameters<"astro:config:setup">["updateConfig"]} params.updateConfig
+			 *
+			 * @see https://astro-integration-kit.netlify.app/utilities/add-vite-plugin/
+			 *
+			 * @example
+			 * ```ts
+			 *  addVitePlugin(plugin)
+			 * ```
+			 */
+			addVitePlugin: (plugin: import("vite").Plugin) => void;
+			/**
+			 * Checks whether an integration is already installed.
+			 *
+			 * @param {object} params
+			 * @param {string} params.name
+			 * @param {config} params.config
+			 *
+			 * @returns {boolean}
+			 *
+			 * @see https://astro-integration-kit.netlify.app/utilities/has-integration/
+			 *
+			 * @example
+			 * ```ts
+			 *  hasIntegration("@astrojs/tailwind")
+			 * ```
+			 */
+			hasIntegration: (name: string) => boolean;
+			/**
+			 * In development, will reload the Astro dev server if any files within
+			 * the integration directory has changed.
+			 *
+			 * @param {string} dir - This is the directory you want to watch for changes
+			 *
+			 * @see https://astro-integration-kit.netlify.app/utilities/watch-integration/
+			 *
+			 * @example
+			 * ```ts
+			 * watchIntegration(resolve())
+			 * ```
+			 */
+			watchIntegration: (dir: string) => void;
+		}
+	>;
+	"astro:config:done"?: AddParam<Hooks["astro:config:done"]>;
+	"astro:server:setup"?: AddParam<Hooks["astro:server:setup"]>;
+	"astro:server:start"?: AddParam<Hooks["astro:server:start"]>;
+	"astro:server:done"?: AddParam<Hooks["astro:server:done"]>;
+	"astro:build:start"?: AddParam<Hooks["astro:build:start"]>;
+	"astro:build:setup"?: AddParam<Hooks["astro:build:setup"]>;
+	"astro:build:generated"?: AddParam<Hooks["astro:build:generated"]>;
+	"astro:build:ssr"?: AddParam<Hooks["astro:build:ssr"]>;
+	"astro:build:done"?: AddParam<Hooks["astro:build:done"]>;
+}
+
+// biome-ignore lint/complexity/noBannedTypes: we know what we are doing
```

Comment:
yeahhh shut up

Reply:
Lovely comment, but I do believe the appropriate value here should be `Record<never, never>` which is slightly different from `{}` in general (although it would have the exact same behavior in this particular case).

Reply:
This does not do the same here. What you suggest completely overrides the initial params